src/deps_compat.jl

```diff
@@ -1,84 +1,73 @@
 """
-    Aqua.test_deps_compat(package; [ignore])
+    Aqua.test_deps_compat(package)
 
-Test that `Project.toml` of `package` list all `compat` for `deps`.
+Test that the `Project.toml` of `package` lists a `compat` entry for each `deps`.
 
 # Arguments
 - `packages`: a top-level `Module`, a `Base.PkgId`, or a collection of
   them.
 
 # Keyword Arguments
+- `broken::Bool = false`: If true, it uses `@test_broken` instead of
+  `@test`.
 - `ignore::Vector{Symbol}`: names of dependent packages to be ignored.
 """
-function test_deps_compat(packages; kwargs...)
-    @testset "$(result.label)" for result in analyze_deps_compat(packages; kwargs...)
-        @debug result.label result
-        @test result ⊜ true
+function test_deps_compat(pkg::PkgId; broken::Bool = false, kwargs...)
+    result = find_missing_deps_compat(pkg; kwargs...)
+    if broken
+        @test_broken result == []
```

Comment:
Maybe do this here and in multiple other places? It avoids allocating a temporary vector. Then again performance doesn't really matter here, so... *shrug*
```suggestion
        @test_broken isempty(result)
```

Reply:
I'll do that in this PR for this tests and something analogous for the rest in a future PR.